packages/nimble-components/src/wafer-map/tests/wafer-map.spec.ts

```diff
@@ -116,6 +118,61 @@ describe('WaferMap', () => {
         });
     });
 
+    describe('worker renderer draw wafer action', () => {
+        let matrixRendererSpy: jasmine.SpyObj<Remote<MatrixRenderer>>;
+        let setupWaferSpy: jasmine.Spy;
+        beforeEach(() => {
+            setupWaferSpy = spyOn(
+                element.workerRenderer,
+                'setupWafer'
+            ).and.returnValue(Promise.resolve());
+            matrixRendererSpy = jasmine.createSpyObj<Remote<MatrixRenderer>>(
+                'Remote',
+                ['setTransformConfig', 'drawWafer', 'drawText']
+            );
+            matrixRendererSpy.setTransformConfig.and.returnValue(
+                Promise.resolve()
+            );
+            matrixRendererSpy.drawWafer.and.returnValue(Promise.resolve());
+            matrixRendererSpy.drawText.and.returnValue(Promise.resolve());
+            element.workerRenderer.matrixRenderer = matrixRendererSpy;
+        });
+
+        it('will call setupWafer and drawWafer but not drawText when labels are hidden', async () => {
+            element.dieLabelsHidden = true;
+            element.canvasWidth = 500;
+            element.canvasHeight = 500;
+            element.diesTable = tableFromArrays({
+                colIndex: Int32Array.from([]),
+                rowIndex: Int32Array.from([]),
+                value: Float64Array.from([])
+            });
+            await waitForUpdatesAsync();
+            await waitForUpdatesAsync();
```

Comment:
These double waitForUpdatesAsync have me a bit concerened. What is the timing that is happening here? Can you explain it?

Reply:
This happens because I create a lot of mock async methods. the latest one which is draw text does not get called in the specified timeout

Reply:
So to give an example, for normal component work when we update an attribute, etc, that work is done asynchronously in an update queue and `waitForUpdatesAsync` is a way to wait for that queue to empty. It's not really being used as a timing function, it's knowing that specific work was added to the queue and we know it will complete. We "wait for the [DOM] updates queue to be emptied". That makes tests reliable, they are based on some specific thing we know is happening, not based on timing (ideally).

What I want to know is if this is reliable or if this is timing based and up to the worker running quickly or slowly (worker or whatever happens to take the time of 2 waitForAsync). If this is timing based then we should instead make a helper, etc, that can reliably wait for something, i.e. like a `waitForExperimentalUpdate()` helper that can `await waferMap.currentTask` or something.

We want to wait for a specific thing, one cycle of the task queue, one cycle of the wafer task, etc. instead of just 16ms or 16ms * 2 which may result in intermittent tests, etc.

Reply:
I changed the second await for the current task and this seems to work. the two waitForUpdatesAsync were working because the first one was setting the current task and the second one was awaiting it when finding it was not undefined